### ISSUE TITLE
Feeback popup shows incorrect amount of available characters(twitter raised the limit today)

### ISSUE THREAD

**Issue by Jasius** [Microsoft/vscode#35214]
- VSCode Version: Code 1.16.1 (27492b6bf3acb0775d82d2f87b25a93490673c6d, 2017-09-14T16:38:23.027Z)
- OS Version: Windows_NT x64 10.0.15063
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|0.1.3
Bookmarks|ale|0.16.0
project-manager|ale|0.21.1
ng-template|Ang|0.1.7
npm-intellisense|chr|1.3.0
path-intellisense|chr|1.4.2
open-html-in-browser|cod|0.1.21
githistory|don|0.2.3
gitlens|eam|5.3.0
auto-rename-tag|for|0.0.14
beautify|Hoo|1.1.1
vscode-git-tags|how|1.1.0
Go|luk|0.6.65
gitignore|mic|1.0.1
prettify-json|moh|0.0.3
vscode-versionlens|pfl|0.19.1
code-settings-sync|Sha|2.8.3
react-beautify|tai|0.3.0
vscode-wakatime|Wak|1.1.16

(1 theme extensions excluded)

---

Steps to Reproduce:

1. Click smiley icon(feedback) at the bottom right
2. Observe that it shows that you have 117 characters available

With new limit you should have 257 characters available, here's the blog post https://blog.twitter.com/official/en_us/topics/product/2017/Giving-you-more-characters-to-express-yourself.html

Here's a screenshot in case It's not clear what I am referring to.
![image](https://user-images.githubusercontent.com/11031051/30906278-11d06966-a380-11e7-8aa4-7fe330d29b00.png)

Reproduces without extensions: Yes

**Comment by bpasero** [comment]
The limit was not yet lifted for everyone.

**Comment by bpasero** [comment]
Also, the limit only seems to be 280 for non-chinese, non-japanese input. So I am not sure how to best change this actually unless we can detect the language of the user.

**Comment by Jasius** [comment]
> unless we can detect the language of the user.

Yeah that's an issue, I guess you could show the limit based on VSC locale as it is available in Chinese, Japanese https://code.visualstudio.com/docs/getstarted/locales

It wouldn't be perfect as in the end it would depend on selected Twitter account language*(I assume it's based on that rather than characters that you input)*

**Comment by kumarharsh** [comment]
@Jasius - yeah, I thought so too but apparently twitter's logic has become very convoluted. https://github.com/Microsoft/vscode/pull/35224#discussion_r151204969

**Comment by Jasius** [comment]
I meant to utilize VSC locale :), checking for twitter locale would be too troublesome for such a minor feature.

**Comment by kumarharsh** [comment]
No, i understood you, but it's not dependent on locale at all. Its dependent on the cgaracters you use. So, even a Chinese locale, when typing in English, can give you 280 chars. And even in the en-US locale, if you type English, but use double-width characters, then you'll only get 140 characters.

**Comment by chrbala** [comment]
It's not actually Twitter locale that is used as the counter. Each character is counted separately, so mixed strings have a more ambiguous number of characters allowed. Each string has a sort of 'score' associated with it, and the score can be no higher than 280.

'abc' -> 3 points
'あああ' -> 6 points
'😀😀😀' -> 6 points
'あbc' -> 4 points
'😀bc' -> 4 points

You could implement the counter with logic like this, where characters are 2 points by default and can be whitelisted to be other point values. It seems that this would be a conservative route that leans toward false positives (strings marked too long, when they are actually okay).

**Comment by kumarharsh** [comment]
There are a bunch of other considerations too, such as URL parsing, etc, which would need to be handled. Also, given how their `twitter-text` API is designed (see link in my comment above), I get a feeling that Twitter might be open to experiment a little with their algorithm for counting letters or the character limit.
